src/arrayMethodJoin.js

```diff
@@ -6,6 +6,55 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+    if (typeof separator === 'number' && !isNaN(separator)) {
+      return this;
+    }
+
+    const stringConsctuctor = (separatorSymbol) => {
+      let result = '';
+
+      for (let i = 0; i < this.length; i++) {
+        const currentElement = this[i];
+        const isLastElement = i === this.length - 1;
+        const isElementValid = currentElement === undefined
+          || currentElement === null
+          || currentElement === '';
+
+        if (isLastElement) {
+          result += `${currentElement}`;
+        } else if (isElementValid) {
+          result += `INVALID_PROPERTY${separatorSymbol}`;
```

Comment:
Why do you need to mark with `INVALID_PROPERTY` and then replace with empty space ?